pkg/build/docker.go

```diff
@@ -33,9 +33,9 @@ func Docker(manifest model.Manifest) error {
 		env = append(env, "ISTIO_ENVOY_BASE_URL="+manifest.ProxyOverride)
 	}
 
-	// Istio operator requires compiled-in charts to be generated before the image is built.
-	if err := util.RunMake(manifest, "istio", env, "gen-charts"); err != nil {
-		return fmt.Errorf("failed to make istio gen-charts: %v", err)
+	// Make sure operator's compiled-in charts are up to date before the image is built.
+	if err := util.RunMake(manifest, "istio", env, "gen-charts", "check-clean-repo"); err != nil {
```

Comment:
won't check-clean-repo fail after `gen-charts`?

Reply:
that's the idea :)

Reply:
And this failure is legit, until we fix it in istio/istio (in 1.7 branch)